stingray/lightcurve.py

```diff
@@ -246,10 +272,7 @@ def __init__(
             time, counts, err = self.initial_optional_checks(time, counts, err, gti=gti)
 
         if time.size != counts.size:
-            raise StingrayError("time and counts array are not " "of the same length!")
-
-        if time.size <= 1:
-            raise StingrayError("A single or no data points can not create " "a lightcurve!")
+            raise StingrayError("time and counts array are not of the same length!")
```

Comment:
I think this might need an additional message for the failure mode where the user puts in a time array but not a counts array. I could imagine this happening if someone thinks they can generate a light curve from time-tagged arrivals by invoking `Lightcurve(time=toas)` rather than using the `Lightcurve.make_lightcurve()` class method. I would extend the error message here to include that for constructing a `Lightcurve()` object, but `time` and `counts` are required, and that for generating a Light curve from TOAs, use the class method

Reply:
Good point. I changed a little bit the machinery, so that the existing error on the invalid counts array is more informative, and also includes the case where counts is None.